### PR TITLE
Support determining FQDN via /etc/myname

### PR DESCRIPTION
**Pull request by ryanakca**
OpenBSD hosts configure their hostname via the /etc/myname file.
This file should contain a single line that is the host's FQDN.
See myname(5) for details.

### CODE REVIEW

File: pkg/chezmoi/chezmoi_unix.go [hunk 1]
```diff
@@ -49,6 +52,23 @@ func etcHostnameFQDNHostname(fileSystem vfs.FS) (string, error) {
...
 
+// etcMynameFQDNHostname returns the FQDN hostname from parsing /etc/myname.
+// See OpenBSD's myname(5) for details on this file.
+func etcMynameFQDNHostname(fileSystem vfs.FS) (string, error) {
+	contents, err := fileSystem.ReadFile("/etc/myname")
+	if err != nil {
+		return "", err
+	}
+	s := bufio.NewScanner(bytes.NewReader(contents))
+	for s.Scan() {
```
Comment (twpayne) [inline pkg/chezmoi/chezmoi_unix.go:63]: This loop needs to ignore comment lines as well.

File: pkg/chezmoi/chezmoi_unix.go [hunk 2]
```diff
@@ -26,6 +26,9 @@ func FQDNHostname(fileSystem vfs.FS) string {
 	if fqdnHostname, err := etcHostsFQDNHostname(fileSystem); err == nil && fqdnHostname != "" {
 		return fqdnHostname
 	}
+	if fqdnHostname, err := etcMynameFQDNHostname(fileSystem); err == nil && fqdnHostname != "" {
```
Comment (twpayne) [inline pkg/chezmoi/chezmoi_unix.go:29]: As only OpenBSD uses this, could you please put this inside an `if runtime.GOOS == "openbsd" { ... }` conditional.

**Review by twpayne — COMMENTED** [review]
Looks great! Just a couple of small nits.

File: pkg/chezmoi/chezmoi_unix.go (hunk 1 above)
Comment (ryanakca) [reply pkg/chezmoi/chezmoi_unix.go:66]: Done!

File: pkg/chezmoi/chezmoi_unix.go (hunk 2 above)
Comment (ryanakca) [reply pkg/chezmoi/chezmoi_unix.go]: Done as well!

**Comment by twpayne** [comment]
Thank you very much :)

**Pull request merged**